src/wasm/apis/sys.rs

```diff
@@ -1,29 +1,15 @@
-use std::collections::HashMap;
-
-use matrix_sdk::{
-    ruma::{OwnedRoomAliasId, OwnedRoomId},
-    Client,
-};
-
 use crate::wasm::GuestState;
 
 wit_bindgen_host_wasmtime_rust::generate!({
     import: "./wit/sys.wit",
     name: "sys"
 });
 
-pub(super) struct SysApi {
-    client: Client,
-    /// In-memory cache for the room alias to room id mapping.
-    room_cache: HashMap<OwnedRoomAliasId, OwnedRoomId>,
-}
+pub(super) struct SysApi;
 
 impl SysApi {
-    pub fn new(client: Client) -> Self {
-        Self {
-            client,
-            room_cache: Default::default(),
-        }
+    pub fn new() -> Self {
```

Comment:
we could remove this ctor now

Reply:
Sounds good! I left it in because `rustc` emits a warning when it's removed, but I'll remove it (lmk if you change your mind).

Reply:
Nevermind, just got confused by my editor and needed to remove a call to `new`.